File: Algorithms/src/1267. Count Servers that Communicate/Code/solution.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        rows, cols = len(grid), len(grid[0])
        rowCount = [0] * rows
        colCount = [0] * cols
        
        # First pass: Count servers in each row and column
        for i in range(rows):
            for j in range(cols):
                if grid[i][j] == 1:
                    rowCount[i] += 1
                    colCount[j] += 1
        
        # Second pass: Count communicable servers
        count = 0
        for i in range(rows):
            for j in range(cols):
                if grid[i][j] == 1 and (rowCount[i] > 1 or colCount[j] > 1):
                    count += 1
        
        return count

File: Algorithms/src/1267. Count Servers that Communicate/Code/solution.cpp
class Solution
{
public:
    int countServers(vector<vector<int>> &grid)
    {
        int rows = grid.size();
        int cols = grid[0].size();
        vector<int> rowCount(rows, 0), colCount(cols, 0);

        // First pass: Count servers in each row and column
        for (int i = 0; i < rows; ++i)
        {
            for (int j = 0; j < cols; ++j)
            {
                if (grid[i][j] == 1)
                {
                    rowCount[i]++;
                    colCount[j]++;
                }
            }
        }

        // Second pass: Count communicable servers
        int count = 0;
        for (int i = 0; i < rows; ++i)
        {
            for (int j = 0; j < cols; ++j)
            {
                if (grid[i][j] == 1 && (rowCount[i] > 1 || colCount[j] > 1))
                {
                    count++;
                }
            }
        }
        return count;
    }
};
